app/models/middleware_server.rb

```diff
@@ -33,6 +37,23 @@ def evaluate_alert(alert_id, event)
   end
 
   def in_domain?
-    !middleware_server_group.nil?
+    middleware_server_group
+  end
+
+  def hawkular?
+    product == 'Hawkular'
+  end
+
+  def immutable?
+    # 'Immutable' key is used in Hawkular servers, while EAP/WildFly servers
+    # use 'Is Immutable' as the key on properties.
+    hawkular? ||
+      ['Immutable', 'Is Immutable'].any? do |key|
```

Comment:
I will confirm that we need to check for both: 'Immutable' and 'Is Immutable' . It has a bad smell (not from anything you have coded but just in general design consistency).